src/security/builtin_plugins/cryptographic/src/crypto_transform.c

```diff
@@ -353,11 +346,11 @@ static bool initialize_remote_session_info(remote_session_info *info, struct sec
   return crypto_calculate_session_key(&info->key, info->id, master_salt, master_key, transformation_kind, ex);
 }
 
-static bool read_subnsg_header(unsigned char **ptr, unsigned char *endp, uint8_t smid, SubmessageHeader_t *hdr, bool *bswap)
+static bool read_submsg_header(unsigned char **ptr, unsigned char *endp, uint8_t smid, SubmessageHeader_t *hdr, bool *bswap)
 {
   SubmessageHeader_t*smhdr = (SubmessageHeader_t *)*ptr;
 
-  if ((uintptr_t)*ptr + RTPS_SUBMESSAGE_HEADER_SIZE > (uintptr_t)endp)
+  if (RTPS_SUBMESSAGE_HEADER_SIZE > (size_t) (endp - *ptr))
```

Comment:
Something I didn't spot the first time round: `octetsToNextHeader` can be garbage, but some of the callers trust it. I think it would be best to check it here (and that unfortunately also means dealing with the case of it being 0), and remove some of then-superfluous checks at some of the call sites.